components/server/src/server.rs

```diff
@@ -153,11 +154,13 @@ pub fn run_tikv(config: TiKvConfig) {
 
     let _m = Monitor::default();
 
-    if !config.raft_engine.enable {
-        run_impl::<RocksEngine>(config)
-    } else {
-        run_impl::<RaftLogEngine>(config)
-    }
+    dispatch_api_version!(config.storage.api_version(), {
```

Comment:
can we remove it now? API is already a generic type.

Reply:
En... remove what ? [`dispatch_api_version`](https://github.com/pingyu/tikv/blob/5757e155a915dbe7387e8cdbab3943e2deb046b0/components/api_version/src/lib.rs#L113) is the macro to match `config.storage.api_version()` and dispatch `api_version::[APIV1, APIV1TTL, APIV2]` as `API`.

Reply:
Oh, `run_tikv` is not generic, forget it